src/features/stories/storiesComponents/BackendAccess.ts

```diff
@@ -23,19 +23,63 @@ export const getStories = async (): Promise<Response | null> => {
 };
 
 export const insertStory = async (
-  githubname: string,
-  filename: string
+  authorId: number,
+  title: string,
+  content: string,
+  pinOrder?: number
 ): Promise<Response | null> => {
   try {
-    const resp = await fetch(`${Constants.storiesBackendUrl}/api/insert_story`, {
+    const resp = await fetch(`${Constants.storiesBackendUrl}/stories`, {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json'
       },
       body: JSON.stringify({
         story: {
-          githubname: githubname,
-          filename: filename
+          authorId: authorId,
+          title: title,
+          content: content,
+          pinOrder: pinOrder
+        }
+      })
+    });
+    if (!resp.ok) {
+      showWarningMessage(
+        `Error while communicating with backend: ${resp.status} ${resp.statusText}${
+          resp.status === 401 || resp.status === 403
+            ? '; try logging in again, after manually saving any work.'
+            : ''
+        }`
+      );
+      return null;
+    }
+    return resp;
+  } catch (e) {
+    showWarningMessage('Error while communicating with backend; check your network?');
+
+    return null;
+  }
+};
+
+export const updateStory = async (
+  id: number,
+  authorId: number,
```

Comment:
You can't update the author ID of a story. Refer to https://github.com/source-academy/stories-backend/blob/main/params/stories/update.go for the allowable fields.